### PR TITLE
Add supply migration for kava-8

### PR DESCRIPTION
**Pull request by karzak**
Adds SWP token to `supply.GenesisState` so that `supply` module invariants pass

### CODE REVIEW

File: migrate/v0_15/migrate.go [hunk 1]
```diff
@@ -258,6 +266,14 @@ func DistributeSwpTokens(genesisState auth.GenesisState) auth.GenesisState {
 	return auth.NewGenesisState(genesisState.Params, accounts)
 }
 
+// MigrateSupply reconciles supply from kava-3 to kava-4
```
Comment (nddeluca) [inline migrate/v0_15/migrate.go:269]: Comment needs to be updated?

File: migrate/v0_15/migrate_test.go [hunk 2]
```diff
@@ -36,6 +36,29 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func TestMigrateFull(t *testing.T) {
```
Comment (nddeluca) [inline migrate/v0_15/migrate_test.go:39]: Nice like this test 👍

File: migrate/v0_15/migrate.go [hunk 3]
```diff
@@ -74,6 +74,14 @@ func MigrateAppState(v0_14AppState genutil.AppMap) {
 		v0_14AppState[auth.ModuleName] = v0_15Codec.MustMarshalJSON(Auth(v0_15Codec, authGenState, GenesisTime))
 	}
 
+	// Migrate supply app state
+	if v0_14AppState[supply.ModuleName] != nil {
+		var supplyGenState supply.GenesisState
+		v0_14Codec.MustUnmarshalJSON(v0_14AppState[supply.ModuleName], &supplyGenState)
+		delete(v0_14AppState, supply.ModuleName)
+		v0_14AppState[supply.ModuleName] = v0_15Codec.MustMarshalJSON(Supply(supplyGenState, sdk.NewCoin("swp", sdk.NewInt(250000000e6))))
```
Comment (nddeluca) [inline migrate/v0_15/migrate.go:82]: 250 million looks correct. @rhuairahrighairidh agree?

Comment (nddeluca) [reply migrate/v0_15/migrate.go:82]: Might be nice to put in a named constant

**Review by nddeluca — COMMENTED** [review]
Looks good, should wait for second look from @rhuairahrighairidh before merge

File: migrate/v0_15/migrate_test.go (hunk 2 above)
Comment (rhuairahrighairidh) [reply migrate/v0_15/migrate_test.go:39]: seconded

**Review by rhuairahrighairidh — APPROVED** [review]
looks good, tested it locally with a genesis export from a week or so ago

**Pull request merged**